openadmet/toolkit/chemoinformatics/activity_funcs.py

```diff
@@ -23,22 +24,23 @@ def calculate_pac50(activity:float, input_unit:str) -> float:
         "nM": unit.nanomolar,
     }
 
-    input_molar = unit_map.get(input_unit)
 
-    if input_molar is None:
-        raise ValueError(f"Unsupported molarity unit: {input_molar}. Must be one of {unit_map.keys()}.")
+    input_unit = unit_map.get(input_unit_str)
 
-    # Get activity with the appropriate molarity units
-    activity_m = (activity * input_molar).to(unit.molar)
+    if input_unit is None:
+        raise ValueError(f"Unsupported molarity unit: {input_unit_str}. Must be one of {unit_map.keys()}.")
 
+    # Get activity with the appropriate molarity units
+    activity_m = (activity * input_unit).to(unit.molar)
     # Now, we can calculate the pAC50 value
     if activity > 0:
         return -np.log10(activity_m.magnitude)
     else:
         raise ValueError("Whoops! Negative activity measure values are not allowed.")
 
 def pac50_to_ki(pac50:float) -> float:
```

Comment:
This is good, but we should expose the full Cheng-Prusoff equation here with the option to leave of the Km and Vmax terms with optional argument. I.E the "assumption" we are making should be optional. 

See here for the maths: https://mtenn.readthedocs.io/en/latest/_autosummary/mtenn.readout.PIC50Readout.html#mtenn.readout.PIC50Readout which you will need to convert units in.


 Also strictly lets use PIC50 here in notation for **this** function as $K_i$ is inhibitory constant (unless we want to work in abstract $K$ units.)
 
 So the signature can look like.

```python
def pIC50_to_Ki(pIC50, S=None, Km=None):
    # warn if S and Km not provided and assume S/Km  is 0 
    # otherwise proceed as before.
```